tests/jballerina-unit-test/src/test/resources/test-src/expressions/literals/numeric_literal_assignment_negative.bal

```diff
@@ -0,0 +1,441 @@
+    Foo5 _ = 4.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 5.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 5.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 4f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 4d;
+    Foo5 _ = 5f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 5d;
+    Foo5 _ = 2.0;
+    Foo5 _ = 2; //error: incompatible types: expected 'Foo5', found 'int'
+    Foo5 _ = 6.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 6.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 2f;
+    Foo5 _ = 2d;
+    Foo5 _ = 6f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 6d;
+    Foo5 _ = 3.0;
+    Foo5 _ = 3; //error: incompatible types: expected 'Foo5', found 'int'
+    Foo5 _ = 7.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 7.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 3f;
+    Foo5 _ = 3d;
+    Foo5 _ = 7f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 7d;
+    Foo5 _ = 1;
+    Foo5 _ = 10; //error: incompatible types: expected 'Foo5', found 'int'
+    Foo5 _ = 1.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 1.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 1f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 1d;
+    Foo5 _ = 10f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 10d;
+    Foo5 _ = 2.1;
+    Foo5 _ = 2.1f;
+    Foo5 _ = 2.1d;
+    Foo5 _ = 8.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 8.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 8d;
+    Foo5 _ = 8f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 9.0; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 9.2; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 9d;
+    Foo5 _ = 9f; //error: incompatible types: expected 'Foo5', found 'float'
+    Foo5 _ = 24; //error: incompatible types: expected 'Foo5', found 'int'
+}
+
+type Bar1 decimal|4d;
+type Bar2 decimal|float;
+type Bar3 decimal|2f;
+type Bar4 decimal|3.0;
+type Bar5 decimal|1;
+type Bar6 decimal|byte;
+type Bar7 decimal|2.1;
+type Bar8 4d|float;
+type Bar9 4d|2f;
+type Bar10 4d|3.0;
+type Bar11 4d|1;
+type Bar12 4d|byte;
+type Bar13 4d|2.1;
+type Bar14 float|2f;
+type Bar15 float|3.0;
+type Bar16 float|1;
+type Bar17 float|byte;
+type Bar18 float|2.1;
+type Bar19 2f|3.0;
+type Bar20 2f|1;
+type Bar21 2f|byte;
+type Bar22 2f|2.1;
+type Bar23 3.0|1;
+type Bar24 3.0|byte;
+type Bar25 3.0|2.1;
+type Bar26 1|byte;
+type Bar27 1|2.1;
+type Bar28 byte|2.1;
+
+function test() {
+    Bar1 _ = 8.0;
```

Comment:
Do all of these have to be tests with separate type definitions? IMO, we can have a few like this and for the others use in-line unions so that's easier to figure out the scenario being tested.

E.g.,
```
decimal|4d _ = 8.0;
```